### ISSUE TITLE
Docker: Build with Buildkit fails

### ISSUE THREAD

**Issue by ilstefano** [gitpod-io/gitpod#2681]
### Describe the bug
When using buildkit ```docker build``` exits with 
```failed to solve with frontend dockerfile.v0: failed to solve with frontend gateway.v0: runc did not terminate sucessfully```

### Steps to reproduce
1. enable Feature Preview in workspace
2. run '''sudo docker-up''' in terminal 1
3. in terminal 2 create Dockerfile: ```echo "# syntax=docker/dockerfile:experimental" > Dockerfile```
3. run ```DOCKER_BUILDKIT=1 docker build -f Dockerfile .```

### Expected behavior
An image should be created or at least docker should complain about an empty Dockerfile.
Remark: I also tested valid Dockerfiles (e.g. with ```FROM alpine``` as line 2). The error stays the same.

### Additional information
Log: 
```bash
gitpod /workspace/buildkit $ echo "# syntax=docker/dockerfile:experimental" > Dockerfile
gitpod /workspace/buildkit $ DOCKER_BUILDKIT=1 docker build -f Dockerfile .
[+] Building 1.0s (4/4) FINISHED                                                                                                                                          
 => [internal] load .dockerignore                                                                                                                                    0.0s
 => => transferring context: 2B                                                                                                                                      0.0s
 => [internal] load build definition from Dockerfile                                                                                                                 0.0s
 => => transferring dockerfile: 85B                                                                                                                                  0.0s
 => resolve image config for docker.io/docker/dockerfile:experimental                                                                                                0.8s
 => CACHED docker-image://docker.io/docker/dockerfile:experimental@sha256:600e5c62eedff338b3f7a0850beb7c05866e0ef27b2d2e8c02aa468e78496ff5                           0.0s
failed to solve with frontend dockerfile.v0: failed to solve with frontend gateway.v0: runc did not terminate sucessfully
```

### Example repository
Workspace: dde895ef-c212-4205-b736-f9ec81eba2fc
Repo: https://github.com/ilstefano/buildkit

**Comment by csweichel** [comment]
I reckon this might have a root cause similar to #2123

**Comment by csweichel** [comment]
This issue was fixed in #3409 and will be deployed soon.
